tests/test_helper.rb

```diff
@@ -79,5 +79,19 @@ def append_file(name, contents)
       f.puts contents
     end
   end
-  
+
+  # Runs a block inside an environment with customized ENV variables.
+  # It restores the ENV after execution.
+  #
+  # @param [Proc] block block to be executed within the customized environment
+  #
+  def with_custom_env_variables(&block)
+    saved_env = {}
+    begin
+      Git::Lib::ENV_VARIABLE_NAMES.each { |k| saved_env[k] = ENV[k] }
+      return block.call
+    ensure
+      Git::Lib::ENV_VARIABLE_NAMES.each { |k| ENV[k] = saved_env[k] }
+    end
+  end
```

Comment:
(This reminds me about my dislike for `test-unit`; with `rspec`, this could easily be done using an `around {}` block instead.)

Reply:
I prefer `rspec` for the structure it makes possible.  I suspect that others don't care for `rspec` because that structure comes with additional complexity and a steeper learning curve.

That said, I respect any project that has tests no matter what framework!

Reply:
Comparing `rspec` to other test frameworks in general, like `testng` in Java, I think the fact that you can actually "group" and structure your tests with `rspec` is a big boon. It's one of the things I'm missing from no longer using it actually.

If you feel like rewriting the tests to use `rspec`, I clearly wouldn't object, but this is definitely not an obligation in any way. 😃